Source/WebGPU/WebGPU/RenderPassEncoder.mm

```diff
@@ -118,8 +143,14 @@
     if (m_vertexDynamicOffsets.size())
         [m_renderCommandEncoder setVertexBytes:&m_vertexDynamicOffsets[0] length:m_vertexDynamicOffsets.size() * sizeof(m_vertexDynamicOffsets[0]) atIndex:m_device->maxBuffersPlusVertexBuffersForVertexStage()];
 
-    if (m_fragmentDynamicOffsets.size())
-        [m_renderCommandEncoder setFragmentBytes:&m_fragmentDynamicOffsets[0] length:m_fragmentDynamicOffsets.size() * sizeof(m_fragmentDynamicOffsets[0]) atIndex:m_device->maxBuffersForFragmentStage()];
+    ASSERT(m_fragmentDynamicOffsets.size());
+    static_assert(sizeof(m_fragmentDynamicOffsets[0]) == sizeof(m_minDepth), "expect dynamic offsets container to have matching size to depth values");
+    auto fragmentsLastBufferSize = m_fragmentDynamicOffsets.size();
```

Comment:
Does this need to be explicitly passed to the compiler?

Reply:
Wouldn't it be the maximum value of `BindGroupLayoutEntry::fragmentBufferDynamicOffset.has_value() ? *BindGroupLayoutEntry::fragmentBufferDynamicOffset + 3: 2` in the pipeline layout passed to the compiler?

Reply:
But frag_depth is an output, which BindGroupLayoutEntry would I use? Maybe I could iterate the list the list of entries and find the highest value?

Reply:
That would work but maybe it’s simpler for me to pass it since I already have the index. 

I thought initially this wouldn’t work due to auto layout, but it is not possible to use dynamic offsets with auto layout. So I would just pass 2 in that case I think.